microannotate/generator.py

```diff
@@ -233,9 +233,26 @@ def __init__(
 
         async with aiofiles.open(os.path.join(self.repo.workdir, path), "wb") as f:
             if self.tokenize_enabled:
-                await f.writelines(
-                    word.group(0) + b"\n" for word in SPLIT_WORD_REGEX.finditer(content)
-                )
+                if not path.endswith(".py"):
+                    lines = (
+                        word.group(0) + b"\n"
+                        for word in SPLIT_WORD_REGEX.finditer(content)
+                    )
+                else:
+                    # In Python files, whitespaces are meaningful when used for defining
+                    # indentation, so we should not ignore them.
+                    lines = []
+                    words = list(SPLIT_WORD_REGEX.finditer(content))
+                    for i, word in enumerate(words):
```

Comment:
nit: iterate on
```python
zip(words[:-1], words[1:])
```
and just add lines.append(words[-1].group(0)...) for the last element

Reply:
No longer applies after 9fc4d53.